ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -208,6 +213,58 @@ private[kvutils] case class ProcessTransactionSubmission(
           buildRejectionLogEntry(RejectionReason.Disputed("DuplicateKey: Contract Key not unique")))
     } yield r
 
+  /** Check that all parties mentioned in a transaction exist. */
+  private def validateParties: Commit[Unit] = {
+
+    def foldPartiesInTx[T](tx: GenTransaction.WithTxValue[NodeId, TContractId], z: T)(f: (T, String) => T): T =
+      foldPartiesInTxActors(tx, foldPartiesInTxVals(tx, z)(f))(f)
+
+    def foldPartiesInTxActors[T](tx: GenTransaction.WithTxValue[_, _], z: T)(f: (T, String) => T): T =
```

Comment:
This function doesn't seem to do what it says in the name (stakeholders are not the actors of a create) and it also doesn't handle fetch nodes.

Reply:
How about `foldParties(tx)`, `foldInvolvedParties(tx)`, `foldValueParties(tx)` and `foldParties(v)`? Also agree about adding `NodeFetch` as we came to the conclusion that transactions (of any kind) shouldn't refer to unallocated parties at all.

Reply:
Actually `foldParties(tx)` -> `foldAllParties(tx)` (due to clash otherwise)